src/libraries/Common/tests/TestUtilities/System/IO/PathGenerator.cs

```diff
@@ -17,6 +18,22 @@ public static string GenerateTestFileName([CallerMemberName] string memberName =
                 memberName ?? "TestBase",
                 lineNumber,
                 index.GetValueOrDefault(),
-                Guid.NewGuid().ToString("N").Substring(0, 8)); // randomness to avoid collisions between derived test classes using same base method concurrently
+                GenerateRandomFileSafeString(8)); // randomness to avoid collisions between derived test classes using same base method concurrently
+
+        private static string GenerateRandomFileSafeString(int length)
+        {
+            // A little more entropy than Guid.NewGuid().ToString("N").Substring(0, length))
```

Comment:
For my own education, can you explain what's the advantage of this method over the Guid one?

Reply:
Guid is Hex so only A-F and 0-9